### ISSUE TITLE
libvirt plugin does not report value grouped by vm

### ISSUE THREAD

**Issue by redmask11** [intelsdi-x/snap-plugin-collector-libvirt#20]
##### ISSUE TYPE

  Bug Report

##### SNAP VERSION
root@controller:~/snap/demo-snap# snapd --version
snapd version v0.13.0-beta

##### SNAP PLUGIN VERSION
root@controller:~# snapctl plugin list
NAME             VERSION         TYPE            SIGNED          STATUS          LOADED TIME
libvirt          5               collector       false           loaded          Thu, 28 Apr 2016 21:33:35 EDT

##### CONFIGURATION
NO

##### OS / ENVIRONMENT
root@controller:~# cat /etc/issue
Ubuntu 14.04 LTS \n \l

##### SUMMARY

when user collect metric data for several vm on one server, the namespace of output metrics
is duplicated for one vm, the ideal output is different metrics for each vm 
##### STEPS TO REPRODUCE
1 start snap agent with : snapd -t 0 -l 1

2 load plugin with snapctl:
snapctl plugin load snap-plugin-collector-libvirt
snapctl plugin load snap-publisher-file

3 start task with task file(libvirt.json):
{
"version": 1,
"schedule": {
"type": "simple",
"interval": "10s"
},
"workflow": {
"collect": {
"metrics": {
"/libvirt/controller/*/cpu/cputime":{},
"/libvirt/controller/*/mem/actual_balloon":{},
"/libvirt/controller/*/mem/available":{},
"/libvirt/controller/*/mem/major_fault":{},
"/libvirt/controller/*/mem/max":{},
"/libvirt/controller/*/mem/mem":{},
"/libvirt/controller/*/mem/min_fault":{},
"/libvirt/controller/*/mem/nr":{},
"/libvirt/controller/*/mem/rss":{},
"/libvirt/controller/*/mem/swap_in":{},
"/libvirt/controller/*/mem/swap_out":{},
"/libvirt/controller/*/mem/unused":{}
},
"publish": [
{
"plugin_name": "file",
"config": {
"file": "/tmp/publish"
}
}
]
}
}
}
snapctl task create -t libvirt.json
 
4 check vm list:
root@controller:~# virsh list --all
Id Name State
2 instance-00000022 running
5 instance-0000002d running

5 watch task with snapctl:
snapctl task watch 006f7e99-4017-4c51-ad2c-698795027d0b
Watching Task (006f7e99-4017-4c51-ad2c-698795027d0b):
NAMESPACE DATA TIMESTAMP SOURCE
/libvirt/controller/instance-0000002d/cpu/cputime 14210000000 2016-04-27 01:13:59.218062575 -0400 EDT controller
/libvirt/controller/instance-0000002d/cpu/cputime 3108910000000 2016-04-27 01:13:59.220301598 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/actual_balloon 4194304 2016-04-27 01:13:59.231300627 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/actual_balloon 65536 2016-04-27 01:13:59.226170392 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/available 0 2016-04-27 01:13:59.237051842 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/available 0 2016-04-27 01:13:59.242251192 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/major_fault 696 2016-04-27 01:13:59.252869535 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/major_fault 0 2016-04-27 01:13:59.247979223 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/max 65536 2016-04-27 01:13:59.186531282 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/max 4194304 2016-04-27 01:13:59.188725875 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/mem 4194304 2016-04-27 01:13:59.193724255 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/mem 65536 2016-04-27 01:13:59.191489805 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/min_fault 478474 2016-04-27 01:13:59.204687113 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/min_fault 0 2016-04-27 01:13:59.199485476 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/nr 0 2016-04-27 01:13:59.161261599 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/nr 0 2016-04-27 01:13:59.155908468 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/rss 0 2016-04-27 01:13:59.172605889 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/rss 104808 2016-04-27 01:13:59.167176599 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/swap_in 0 2016-04-27 01:13:59.210279145 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/swap_in 0 2016-04-27 01:13:59.215287744 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/swap_out 0 2016-04-27 01:13:59.183739084 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/swap_out 0 2016-04-27 01:13:59.178371876 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/unused 0 2016-04-27 01:13:59.258578891 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/unused 0 2016-04-27 01:13:59.2635737 -0400 EDT controller

##### EXPECTED RESULTS
all metrics should be collected and displayed for each vm.
for example: for mem/unused. it should be display 
/libvirt/controller/instance-0000002d/mem/unused 0 2016-04-27 01:13:59.258578891 -0400 EDT controller
/libvirt/controller/instance-00000022/mem/unused 0 2016-04-27 01:13:59.2635737 -0400 EDT controller
##### ACTUAL RESULTS
duplicated display for one vm, in my case it's instance-0000002d.
/libvirt/controller/instance-0000002d/mem/unused 0 2016-04-27 01:13:59.258578891 -0400 EDT controller
/libvirt/controller/instance-0000002d/mem/unused 0 2016-04-27 01:13:59.2635737 -0400 EDT controller
@intelsdi-x/plugin-maintainers

**Comment by jcooklin** [comment]
@redmask11: After you load the plugin will you drop also drop the results from `snapctl metric list` here?

Thanks,
Joel

**Comment by sandlbn** [comment]
@redmask11 Here u can find filtered list output 

https://gist.githubusercontent.com/sandlbn/887e8799730938780dfe97838b2336e0/raw/14d875a3eb38a53bdfb48e642479bc554588d74b/metric_list

and this is my task manifest 

https://gist.githubusercontent.com/sandlbn/90f298b0116db3319a0c3922cedc4639/raw/d878ccd4a7c72d877527bca36248479cd2b6fc11/task_manifest.json

Here is my output from task watch 

```
./snap-bin/snap-v0.13.0-beta/bin/snapctl task watch e6f20821-8192-4ec5-a257-eef8da0427e0
Watching Task (e6f20821-8192-4ec5-a257-eef8da0427e0):
NAMESPACE 								 DATA 		 TIMESTAMP 					 SOURCE
/libvirt/irild016.cslf1.intel.com/ubuntu_mesos/cpu/cputime 		 4010610000000 	 2016-05-02 09:52:42.550604766 +0100 IST 	 irild016.cslf1.intel.com
/libvirt/irild016.cslf1.intel.com/ubuntu_mesos/mem/available 		 0 		 2016-05-02 09:52:42.554534676 +0100 IST 	 irild016.cslf1.intel.com
/libvirt/irild016.cslf1.intel.com/ubuntu_mesos/mem/max 			 524288 	 2016-05-02 09:52:42.55546114 +0100 IST 	 irild016.cslf1.intel.com
/libvirt/irild016.cslf1.intel.com/ubuntu_mesos/mem/rss 			 578684 	 2016-05-02 09:52:42.558327968 +0100 IST 	 irild016.cslf1.intel.com
/libvirt/irild016.cslf1.intel.com/vagrant_operator/cpu/cputime 		 1563320000000 	 2016-05-02 09:52:42.5372707 +0100 IST 		 irild016.cslf1.intel.com
/libvirt/irild016.cslf1.intel.com/vagrant_operator/mem/available 	 0 		 2016-05-02 09:52:42.542385688 +0100 IST 	 irild016.cslf1.intel.com
/libvirt/irild016.cslf1.intel.com/vagrant_operator/mem/max 		 4194304 	 2016-05-02 09:52:42.543813228 +0100 IST 	 irild016.cslf1.intel.com
/libvirt/irild016.cslf1.intel.com/vagrant_operator/mem/rss 		 1803868 	 2016-05-02 09:52:42.549528816 +0100 IST 	 irild016.cslf1.intel.com
Stopping task watch
```

Below u can find output from tail -100 /tmp/published 

https://gist.githubusercontent.com/sandlbn/0bca849fd3f4ceec47ccb24120606f62/raw/899c1724b82dfdd2c39150cf8c96ea42a6451941/tmp_published

I tested on Ubuntu 14.04.3 with kernel 

``Linux irild016.cslf1.intel.com 3.19.0-56-generic #62~14.04.1-Ubuntu SMP Fri Mar 11 11:03:15 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux``` with snap version beta 0.13 https://github.com/intelsdi-x/snap/releases/tag/v0.13.0-beta

I also made a test on Fedora 23 with the same results. Everything is working OK for me.

**Comment by sandlbn** [comment]
@redmask11 Did you tried to run a task with my task manifest ?

**Comment by sandlbn** [comment]
@redmask11 any news ?

**Comment by sandlbn** [comment]
Looks like this one can be also related to the #23 .

**Comment by sandlbn** [comment]
Fixed in #24
